apps/api/src/secret/secret.service.ts

```diff
@@ -239,6 +239,32 @@ export class SecretService {
     return secret
   }
 
+  async bulkCreateSecrets(
+    user: AuthenticatedUser,
+    projectSlug: string,
+    secrets: CreateSecret[]
+  ): Promise<SecretWithValues[]> {
+    this.logger.log(
+      `User ${user.id} initiated bulk creation of ${secrets.length} secrets in project ${projectSlug}`
+    )
+
+    const createdSecrets: SecretWithValues[] = []
+
+    for (const secret of secrets) {
+      try {
+        const result = await this.createSecret(user, secret, projectSlug)
+        createdSecrets.push(result)
+      } catch (err) {
+        this.logger.error(
```

Comment:
I'm worried about the transactional aspect of this error catching mechanism. Ideally, we shouldn't ban users from continuing with the secret creation. Let's maintain an array of error strings. If an error occurs, we append to that array